src/test/kotlin/no/nav/syfo/oppfolgingstilfelle/retry/KafkaOppfolgingstilfelleRetrySpek.kt

```diff
@@ -0,0 +1,367 @@
+                oppfolgingstilfelleService = oppfolgingstilfelleService,
+                oppfolgingstilfelleRetryProducer = mockOppfolgingstilfelleRetryProducer
+            )
+
+            val consumerPropertiesOppfolgingstilfelleRetry = kafkaOppfolgingstilfelleRetryConsumerProperties(env, vaultSecrets)
+                .overrideForTest()
+
+            val consumerOppfolgingstilfelleRetry = KafkaConsumer<String, String>(consumerPropertiesOppfolgingstilfelleRetry)
+            consumerOppfolgingstilfelleRetry.subscribe(listOf(oppfolgingstilfelleRetryTopic))
+
+            it("should resend KOversikthendelseRetry to topic when it has not exceeded retry limit and is not ready for retry") {
+                val kOppfolgingstilfelleRetry = generateKOppfolgingstilfelleRetry.copy(
+                    aktorId = ARBEIDSTAKER_AKTORID.aktor,
+                    created = LocalDateTime.now(),
+                    retryTime = LocalDateTime.now().plusHours(1),
+                    retriedCount = 0
+                )
+                val mockConsumerOppfolgingstilfelleRetry = createMockConsumerOppfolgingstilfelleRetry(kOppfolgingstilfelleRetry)
+
+                runBlocking {
+                    pollAndProcessOppfolgingstilfelleRetryTopic(
+                        kafkaConsumer = mockConsumerOppfolgingstilfelleRetry,
+                        oppfolgingstilfelleRetryService = oppfolgingstilfelleRetryService
+                    )
+                }
+
+                val messages = getMessagesOversikthendelsetilfelle(consumerOversikthendelsetilfelle)
+                messages.size shouldBeEqualTo 0
+
+                verify(exactly = 1) { mockOppfolgingstilfelleRetryProducer.sendAgainOppfolgingstilfelleRetry(kOppfolgingstilfelleRetry) }
+            }
+        }
+
+        describe("Read and process KOversikthendelseRetry with unavailable Aktorregister") {
+            val mockOppfolgingstilfelleRetryProducer = mockk<OppfolgingstilfelleRetryProducer>()
+            justRun { mockOppfolgingstilfelleRetryProducer.sendAgainOppfolgingstilfelleRetry(any()) }
+            justRun { mockOppfolgingstilfelleRetryProducer.sendRetriedOppfolgingstilfelleRetry(any()) }
+
+            val oppfolgingstilfelleRetryService = OppfolgingstilfelleRetryService(
+                oppfolgingstilfelleService = oppfolgingstilfelleService,
+                oppfolgingstilfelleRetryProducer = mockOppfolgingstilfelleRetryProducer
+            )
+
+            val consumerPropertiesOppfolgingstilfelleRetry = kafkaOppfolgingstilfelleRetryConsumerProperties(env, vaultSecrets)
+                .overrideForTest()
+
+            val consumerOppfolgingstilfelleRetry = KafkaConsumer<String, String>(consumerPropertiesOppfolgingstilfelleRetry)
+            consumerOppfolgingstilfelleRetry.subscribe(listOf(oppfolgingstilfelleRetryTopic))
+
+            it("should resend KOversikthendelseRetry when retried and failed due to missing Fodselsnummer") {
```

Comment:
Nøyaktig og beskrivende tittel på testen! 

Superpirk: trengs ordet should (gjelder alle testene).

Reply:
Nei, må ikke bruke should.